glib/PointerWrapper.cs

```diff
@@ -93,35 +84,26 @@ static bool PerformQueuedUnrefs ()
 		~PointerWrapper ()
 		{
 			lock (Objects)
-				Objects.Remove (handle);
-
-			if (ObjectDestroyed != null)
-				ObjectDestroyed.Invoke (handle);
+				Objects.Remove (tref.Handle);
 
 			lock (lockObject) {
-				if (tref != null) {
-					PendingDestroys.Add (tref);
-					if (!idle_queued) {
-						Timeout.Add (50, PerformQueuedUnrefsHandler);
-						idle_queued = true;
-					}
+				PendingDestroys.Add (tref);
+				if (!idle_queued) {
+					Timeout.Add (50, PerformQueuedUnrefsHandler);
+					idle_queued = true;
 				}
 			}
 		}
 
 		public void Dispose ()
 		{
-			if (handle == IntPtr.Zero)
+			GC.SuppressFinalize (this);
```

Comment:
Do we care about the perf impact of multiple calls to Dispose resulting in multiple calls to GC.SuppressFinalize? I assume not!

Reply:
Yodo, you only dispose once